### ISSUE TITLE
Fix `declaration-block-no-shorthand-property-overrides` false negatives for `border`

### ISSUE THREAD

**Issue by hudochenkov** [stylelint/stylelint#7506]
### What minimal example or steps are needed to reproduce the bug?

```css
a {
	border-color: red;
	border: 1px solid blue;
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "declaration-block-no-shorthand-property-overrides": true
  }
}
```

### How did you run Stylelint?

All ways.

### Which Stylelint-related dependencies are you using?

```json
"stylelint": "16.2.0"
```

### What did you expect to happen?

To get a violation:

```
Unexpected shorthand "border" after "border-color"
```

### What actually happened?

No violation is reported.

### Do you have a proposal to fix the bug?

Update [shorthands reference](https://github.com/stylelint/stylelint/blob/328544d8dfe4abfb1d75ab6631528f2e3c605a74/lib/reference/properties.mjs#L15).

Looking at the reference, it could be further improved. E. g. `border` doesn't have `border-block`, `border-block-end`, etc. in it's set. Which means that there are more unreported violations.

**Comment by ybiquitous** [comment]
@hudochenkov Thanks for the report. Right, this is a bug of false negatives.

> Update [shorthands reference](https://github.com/stylelint/stylelint/blob/328544d8dfe4abfb1d75ab6631528f2e3c605a74/lib/reference/properties.mjs#L15).

This update only may not be enough. It may be necessary to fix the rule.

> Looking at the reference, it could be further improved. E. g. `border` doesn't have `border-block`, `border-block-end`, etc. in it's set. Which means that there are more unreported violations.

Sounds good. Let's fix them together.

I've labeled the issue as **ready to implement**. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by G-Rath** [comment]
I might have a crack at this

**Comment by G-Rath** [comment]
This is fixed for the given example but not when the property order is switched:

```
a {
  border: 1px solid blue;
  border-color: red;
}
```

While you could argue that might not be a problem of the specifically mentioned rule since it's setting a different value, it also applies when there isn't e.g. a color on `border:

```
a {
  border: 1px solid;
  border-color: red;
}
```
